### ISSUE TITLE
Regression in v0.50.20: metabase fail with `permission denied: "RI_ConstraintTrigger_c_83876" is a system trigger` when using PostgreSQL without superuser rights

### ISSUE THREAD

**Issue by mcimbaro** [Scalingo/metabase-scalingo#20]
**Disclaimer:** It's not an issue on the buildpack side.

The latest metabase version ([v0.50.20](https://github.com/metabase/metabase/releases/tag/v0.50.20), released on Aug 14) fail on Scalingo with the following message:
```java
2024-08-20 16:18:29.947752445 +0200 CEST [web-1] 2024-08-20 14:18:29,879 ERROR liquibase.changelog :: ChangeSet migrations/001_update_migrations.yaml::v50.2024-01-10T03:27:30::noahmoss encountered an exception.
2024-08-20 16:18:29.947759004 +0200 CEST [web-1] liquibase.exception.DatabaseException: ERROR: permission denied: "RI_ConstraintTrigger_c_18139" is a system trigger [Failed SQL: (0) ALTER TABLE data_permissions DISABLE TRIGGER ALL]
2024-08-20 16:18:29.947761031 +0200 CEST [web-1] 	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:179)
2024-08-20 16:18:29.947763847 +0200 CEST [web-1] 	at liquibase.Scope.child(Scope.java:164)
2024-08-20 16:18:29.947773547 +0200 CEST [web-1] 	at metabase.db.liquibase$migrate_up_if_needed_BANG_.invokeStatic(liquibase.clj:349)
2024-08-20 16:18:29.947772956 +0200 CEST [web-1] 	at metabase.db.liquibase$run_in_scope_locked.invokeStatic(liquibase.clj:318)
2024-08-20 16:18:29.947761721 +0200 CEST [web-1] 	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1285)
2024-08-20 16:18:29.947761386 +0200 CEST [web-1] 	at liquibase.executor.AbstractExecutor.execute(AbstractExecutor.java:141)
2024-08-20 16:18:29.947774221 +0200 CEST [web-1] 	at metabase.db.setup$migrate_BANG_$fn__53428.invoke(setup.clj:84)
2024-08-20 16:18:29.947764307 +0200 CEST [web-1] 	at liquibase.Scope.lambda$child$0(Scope.java:186)
2024-08-20 16:18:29.947765022 +0200 CEST [web-1] 	at liquibase.Scope.child(Scope.java:164)
2024-08-20 16:18:29.947775124 +0200 CEST [web-1] 	at metabase.db.setup$migrate_BANG_.invokeStatic(setup.clj:73)
2024-08-20 16:18:29.947763378 +0200 CEST [web-1] 	at liquibase.Scope.child(Scope.java:195)
2024-08-20 16:18:29.947763603 +0200 CEST [web-1] 	at liquibase.Scope.child(Scope.java:185)
2024-08-20 16:18:29.947767991 +0200 CEST [web-1] 	at liquibase.command.CommandScope.execute(CommandScope.java:217)
2024-08-20 16:18:29.947776157 +0200 CEST [web-1] 	at metabase.db.setup$run_schema_migrations_BANG_.invoke(setup.clj:144)
2024-08-20 16:18:29.947769481 +0200 CEST [web-1] 	at liquibase.Scope.child(Scope.java:185)
2024-08-20 16:18:29.947770245 +0200 CEST [web-1] 	at liquibase.Liquibase.runInScope(Liquibase.java:1419)
2024-08-20 16:18:29.947776406 +0200 CEST [web-1] 	at metabase.db.setup$setup_db_BANG_$fn__53456$fn__53457.invoke(setup.clj:167)
2024-08-20 16:18:29.947776640 +0200 CEST [web-1] 	at metabase.util.jvm$do_with_us_locale.invokeStatic(jvm.clj:239)
2024-08-20 16:18:29.947770562 +0200 CEST [web-1] 	at liquibase.Liquibase.update(Liquibase.java:234)
2024-08-20 16:18:29.947768845 +0200 CEST [web-1] 	at liquibase.Scope.lambda$child$0(Scope.java:186)
2024-08-20 16:18:29.947771017 +0200 CEST [web-1] 	at liquibase.Liquibase.update(Liquibase.java:212)
2024-08-20 16:18:29.947777212 +0200 CEST [web-1] 	at metabase.db.setup$setup_db_BANG_$fn__53456.invoke(setup.clj:161)
2024-08-20 16:18:29.947762049 +0200 CEST [web-1] 	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:755)
2024-08-20 16:18:29.947771962 +0200 CEST [web-1] 	at liquibase.Scope.lambda$child$0(Scope.java:186)
2024-08-20 16:18:29.947775343 +0200 CEST [web-1] 	at metabase.db.setup$migrate_BANG_.doInvoke(setup.clj:55)
2024-08-20 16:18:29.947777669 +0200 CEST [web-1] 	at metabase.db.setup$setup_db_BANG_.invoke(setup.clj:153)
2024-08-20 16:18:29.947766975 +0200 CEST [web-1] 	at liquibase.Scope.child(Scope.java:185)
2024-08-20 16:18:29.947767442 +0200 CEST [web-1] 	at liquibase.command.core.AbstractUpdateCommandStep.run(AbstractUpdateCommandStep.java:108)
2024-08-20 16:18:29.947767217 +0200 CEST [web-1] 	at liquibase.Scope.child(Scope.java:164)
2024-08-20 16:18:29.947769164 +0200 CEST [web-1] 	at liquibase.Scope.child(Scope.java:195)
2024-08-20 16:18:29.948075797 +0200 CEST [web-1] 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:371)
2024-08-20 16:18:29.948076331 +0200 CEST [web-1] 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:419)
2024-08-20 16:18:29.948073190 +0200 CEST [web-1] 	at clojure.core$apply.invoke(core.clj:662)
2024-08-20 16:18:29.948074761 +0200 CEST [web-1] 	at metabase.bootstrap.main(Unknown Source)
2024-08-20 16:18:29.948073442 +0200 CEST [web-1] 	at metabase.bootstrap$_main.invokeStatic(bootstrap.clj:31)
2024-08-20 16:18:29.948073667 +0200 CEST [web-1] 	at metabase.bootstrap$_main.doInvoke(bootstrap.clj:28)
2024-08-20 16:18:29.948078105 +0200 CEST [web-1] 	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:464)
2024-08-20 16:18:29.948077126 +0200 CEST [web-1] 	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:302)
2024-08-20 16:18:29.948077790 +0200 CEST [web-1] 	at com.mchange.v2.c3p0.impl.NewProxyStatement.execute(NewProxyStatement.java:75)
2024-08-20 16:18:29.947771477 +0200 CEST [web-1] 	at metabase.db.liquibase$migrate_up_if_needed_BANG_$fn__44483.invoke(liquibase.clj:360)
2024-08-20 16:18:29.947793094 +0200 CEST [web-1] 	at metabase.core$init_BANG__STAR_.invokeStatic(core.clj:117)
2024-08-20 16:18:29.947774443 +0200 CEST [web-1] 	at metabase.db.liquibase$do_with_liquibase$f_STAR___44420.invoke(liquibase.clj:140)
2024-08-20 16:18:29.947774666 +0200 CEST [web-1] 	at metabase.db.liquibase$do_with_liquibase.invokeStatic(liquibase.clj:143)
```

Looking at the changelog, I saw that metabase reworked the way migrations are done: https://github.com/metabase/metabase/issues/46497. Since then, some permissions issues are raised when using PostgreSQL with a non super-user:
- https://github.com/metabase/metabase/issues/46948
- https://github.com/metabase/metabase/issues/46902

The metabase team advised to stick with an oldest version for now.

**Comment by crisptrutski** [comment]
@mcimbaro we have just released 50.21 and the docker images are published

**Comment by mcimbaro** [comment]
> @mcimbaro we have just released 50.21 and the docker images are published

Confirmed, it's fixed! Thanks a lot for the follow up and the quick ship!
